### DISCUSSION TITLE
Accessing iChlor 30 via RS485

### DISCUSSION THREAD

**Discussion by nac1211** [tagyoureit/nodejs-poolController#992] [RS485]
My 6 years old SWG died and need a replacement. I'd like to control the new unit using njsPC integrated with my current home automation software, Homeassistant(HA), ie. setting the Output % as an example. I'm planing on running njsPC as Docker Container on the same hardware that runs HA software with an attached USB-to-RS485 adapter. Integration between HA and njsPC is via MQTT and/or njsPC-HA HACS integration custom component. The new SWG unit I'm looking at is "Pentair iChlor 30" only with it's own power center.  Is this workable solution?  Is iChlor 30 supported hardware by njsPC?   Your feedback is greatly appreciated!!

**Comment by rstrouse** [comment]
How many gallons in your pool and which unit are you replacing?

**Comment by nac1211** [comment]
16,000 Gallons. Replacing current dead Salt Water Chlorine Generator (make: CircuPool RJ30+) with, possibly, Pentair iChlor 30.

**Reply by rstrouse** [reply]
The iChlor 30 should communicate just fine if the cost is close I would pick the IC40 though as I have seen mixed reviews of the iChlor units.  You will need the kit with the power center as this will contain the surge board that communicates with njsPC this is the same board for all the Pentair units up to the IC60.  

You might want to look at using Nixie in njsPC to control your pool.  Especially if you have valves or variable speed pumps.  This will also directly integrate with HA on a meaningful level.

**Comment by nac1211** [comment]
Thank you for your prompt response!! Its good to know iChlor30 will work!! I currently control my pool equipment (two speed pump), SPA, pool lights and Irrigation system using ESP8266 relay boards running ESPHome firmware integrated with Homeassistant. I was looking at iChlor30 since its newer than ICxx models and it costs less, but I will consider an IC40 at close price range. What is the name of the Power Center Kit? I don't have info on Nixie, can you point me to info reqarding that. thanks

**Reply by tagyoureit** [reply]
Power center: https://a.co/d/0ejjmPpE

And nixie wiki: https://github.com/tagyoureit/nodejs-poolController/wiki/Nixie-Controller-Setup

**Comment by nac1211** [comment]
Thank you! 

BTW, I don't see iChlor30 as a model in Dashpanel. Do I pick intelliChlor IC40 as a model?  I don't have a unit to test yet.

Thanks again for your excellent support.

**Reply by rstrouse** [reply]
Yes the iChlor is pretty rare and the protocol is the same.
